docs/source/tutorials/chat.rst

```diff
@@ -0,0 +1,331 @@
+and :code:`<<SYS>>` are tokenized as normal text and not their own IDs.
+
+.. code-block:: python
+
+    print(tokenizer.decode(518))
+    # '['
+    print(tokenizer.decode(25580))
+    # 'INST'
+    print(tokenizer.decode(29962))
+    # ']'
+    print(tokenizer.decode([3532, 14816, 29903, 6778]))
+    # '<<SYS>>'
+
+It's important to note that you should not place the special reserved tokens in your
+input prompts manually, as it will be treated as normal text and not as a special
+token.
+
+.. code-block:: python
+
+    print(tokenizer.encode("<s>", add_bos=False, add_eos=False))
+    # [529, 29879, 29958]
+
+Now let's take a look at LLaMA3's formatting to see how it's tokenized differently
+than LLaMA2.
+
+.. code-block:: python
+
+    from torchtune.modules.tokenizers import TikTokenTokenizer
+
+    tokenizer = TikTokenTokenizer("/tmp/Meta-Llama-3-8B/original/tokenizer.model")
+    messages = [Message.from_dict(msg) for msg in sample]
+    tokens, mask = tokenizer.tokenize_messages(messages)
+    print(tokenizer.decode(tokens))
+    # '<|start_header_id|>system<|end_header_id|>\n\nYou are a helpful, respectful,
+    # and honest assistant.<|eot_id|><|start_header_id|>user<|end_header_id|>\n\nWho
+    # are the most influential hip-hop artists of all time?<|eot_id|><|start_header_id|>
+    # assistant<|end_header_id|>\n\nHere is a list of some of the most influential hip-hop
+    # artists of all time: 2Pac, Rakim, N.W.A., Run-D.M.C., and Nas.<|eot_id|>'
+
+.. note::
+    We used the `tokenize_messages` API for LLaMA3, which is different than
+    encode. It simply manages adding all the special tokens in the correct
+    places after encoding the individual messages.
+
+We can see that the tokenizer handled all the formatting without us specifying a prompt
+template. Turns out, all of the additional tags are special tokens, and we don't require
+a separate prompt template. We can verify this by checking if the tags get encoded
+as their own token IDs.
+
+.. code-block:: python
+
+    print(tokenizer._encode_special_token("<|begin_of_text|>"))
+    # 128000
+    print(tokenizer._encode_special_token("<|eot_id|>"))
+    # 128009
+
+The best part is - all these special tokens are handled purely by the tokenizer.
+That means you won't have to worry about messing up any required prompt templates!
+
+
+When should I use a prompt template?
+------------------------------------
+
+When to use or not to use a prompt template is governed by what your desired inference
+behavior is. You should use a prompt template in the following scenarios:
+
+1. You are running inference on the base model and it was pre-trained with a prompt
+template
+2. You want to prime a fine-tuned model to expect a certain prompt structure on inference
+for a specific task
+
+It is not strictly necessary to fine-tune with a prompt template, but generally you
+want the model to perform some sort of task, which will require some formatting of
+the prompt.
```

Comment:
I might scrap this sentence altogether, esp "you want the model to perform some sort of task" is kinda self-evident